src/neo4j_graphrag/experimental/pipeline/pipeline.py

```diff
@@ -412,6 +416,59 @@ def validate_parameter_mapping_for_task(self, task: TaskPipelineNode) -> bool:
     async def get_final_results(self, run_id: str) -> dict[str, Any]:
         return await self.final_results.get(run_id)  # type: ignore[no-any-return]
 
+    async def stream(self, data: dict[str, Any]) -> AsyncGenerator[Event, None]:
```

Comment:
```python
data: dict[str, Any]
```

I wonder if it's about time we have a data type for pipeline inputs. Or would that be overkill for this PR?

Reply:
We can't have a fixed data type because it depends on the components: SimpleKGPipeline and a RAGPipeline (if we ever add it) won't have the same inputs. I think it makes sense for the "template" pipeline though, and we are close to it with the "SimpleKGPipelineConfig", but for the generic pipeline Idk how we could achieve it.